### PR TITLE
Integrates frontend/backend clients with hello world

### PR DESCRIPTION
**Pull request by a-deal**
This gets us to a comfortable place architecture wise and a functional fullstack setup. I tagged this as a WIP due to having issues getting HMR to play nicely with frontend changes in development mode. As far I can tell, the issue _probably_ lies with `react-hot-loader` and `typescript`.

For more on this, see this [link](https://github.com/gaearon/react-hot-loader#typescript)

Happy to drop HMR and opt for a simpler setup but it would be nice to have. Let me know what you think :)

### CODE REVIEW

File: app/components/Hello.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as React from "react";
```
Comment (thekamahele) [inline app/components/Hello.tsx]: I prefer single quotes - it should be a lint rule already

File: package.json [hunk 2]
```diff
@@ -30,50 +31,57 @@
...
-    "@types/node": "^7.0.18",
+    "@types/node": "^10.9.4",
+    "@types/react": "^16.4.11",
+    "@types/webpack": "^4.4.11",
+    "@types/webpack-dev-server": "^3.1.0",
+    "awesome-typescript-loader": "^5.2.0",
+    "babel-plugin-transform-class-properties": "^6.24.1",
     "concurrently": "^3.4.0",
-    "nodemon": "^1.11.0",
+    "eslint": "^4.10.0",
```
Comment (thekamahele) [inline package.json]: is eslint being used for client side? If we're using typescript across, we may just want to stick to tslint?

**Review by thekamahele — COMMENTED** [review]
Code looks good, need to pull down to test though

**Comment by a-deal** [comment]
HMR is a go!

**Pull request merged**